packages/@romejs/compiler/lint/rules/react/noArrayIndexKey.ts

```diff
@@ -0,0 +1,219 @@
+import {Path, TransformExitResult} from "@romejs/compiler";
+import {
+	JSArrowFunctionExpression,
+	JSBindingIdentifier,
+	JSFunctionExpression,
+} from "@romejs/ast";
+import {descriptions} from "@romejs/diagnostics";
+import {
+	doesNodeMatchPattern,
+	getJSXAttribute,
+	hasJSXAttribute,
+	isFunctionNode,
+} from "@romejs/js-ast-utils";
+
+function getKeyValue(path: Path): string | undefined {
+	let keyValue = undefined;
+	let callExpression = undefined;
+
+	// handle React.cloneElement and cloneElement
+	if (path.node.type === "JSCallExpression") {
+		callExpression = path.node;
+	} else if (path.parent.type === "JSCallExpression") {
+		callExpression = path.parent;
+	}
+
+	if (
+		callExpression &&
+		callExpression.type === "JSCallExpression" &&
+		callExpression.arguments.length > 1 &&
+		callExpression.arguments[1].type === "JSObjectExpression"
+	) {
+		callExpression.arguments[1].properties.forEach((property) => {
+			if (
+				property.type === "JSObjectProperty" &&
+				property.key.type === "JSStaticPropertyKey" &&
+				property.key.value.type === "JSIdentifier" &&
+				property.key.value.name === "key"
+			) {
+				keyValue =
+					property.value.type === "JSReferenceIdentifier" && property.value.name;
+			}
+		});
+	}
```

Comment:
Avoid the usage of `forEach` since we use `for...of` everywhere already.

```suggestion
	if (
		callExpression &&
		callExpression.type === "JSCallExpression" &&
		callExpression.arguments.length > 1
	) {
		const obj = callExpression.arguments[1];
		if (obj.type === "JSObjectExpression") {
			for (const prop of obj) {

				if (
					prop.type === "JSObjectProperty" &&
					prop.key.type === "JSStaticPropertyKey" &&
					prop.key.value.type === "JSIdentifier" &&
					prop.key.value.name === "key"
				) {
					keyValue =
						prop.value.type === "JSReferenceIdentifier" && prop.value.name;
				}
			}
		}
	}
```